### PR TITLE
Introduce set of expressions for URI parts encoding

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: vividus-bdd-engine/src/test/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessorTests.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @ParameterizedTest(name = "{index}: for expression \"{0}\", result is \"{1}\"")
+    @CsvSource({
+        "encodeUrl(/wiki/w3schools), %2Fwiki%2Fw3schools",
+        "encodeUrl(/wls/recipes#!/), %2Fwls%2Frecipes%23%21%2F"
+    })
+    void testExecute(String expression, String expected)
+    {
+        assertEquals(Optional.of(expected), processor.execute(expression));
+        assertThat(logger.getLoggingEvents(), equalTo(List.of(warn(
+                "#{encodeUrl(..)} expression is deprecated, use set of #{encodeUriXyz(..)} expressions instead"))));
```
Comment (ikalinin1) [inline vividus-bdd-engine/src/test/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessorTests.java:58]: Users will try to use it as is `#{encideUriXyz()}` :)

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessor.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+@Named
+@Deprecated(forRemoval = true, since = "0.2.7")
+public class EncodeUrlExpressionProcessor extends UnaryExpressionProcessor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(EncodeUrlExpressionProcessor.class);
+
+    public EncodeUrlExpressionProcessor()
+    {
+        super("encodeUrl", input -> URLEncoder.encode(input, UTF_8));
```
Comment (ikalinin1) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessor.java:40]: Can we just refactor this expression to delegate proper part encoding to a valid method?

File: docs/modules/plugins/pages/plugin-rest-api.adoc [hunk 3]
```diff
@@ -48,3 +48,69 @@ Then server `$hostname` supports secure protocols that $rule `$protocols`
...
+|Encode the given URI query parameter
+
+|`#{encodeUriFragment($fragment)}`
+|Encode the given URI fragment
+|===
+
+==== Examples
+
+|===
+|Expression |Result
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-rest-api.adoc:93]: |Expression
|Result

File: vividus-tests/src/main/resources/story/integration/HTTP.story [hunk 4]
```diff
@@ -22,12 +22,12 @@ When I send HTTP GET to the relative URL '/anything/path-with-&-ampersand'
 Then the response code is equal to '200'
 Then a JSON element by the JSON path '$.url' is equal to '${http-endpoint}anything/path-with-&-ampersand'
 
-Scenario: Verify handling of ampersand character in URL query
-When I send HTTP GET to the relative URL '/get?key=#{encodeUrl(a&b)}'
+Scenario: Verify handling of ampersand and space characters in URL query parameter
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/HTTP.story:25]: I propose to use URI the same as for expression.

File: docs/modules/plugins/pages/plugin-rest-api.adoc (hunk 3 above)
Comment (valfirst) [reply docs/modules/plugins/pages/plugin-rest-api.adoc:93]: it won't work without `options="header"`, so the current approach looks more concise

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessor.java (hunk 2 above)
Comment (valfirst) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessor.java:40]: there is no such valid method :)

File: vividus-bdd-engine/src/test/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessorTests.java (hunk 1 above)
Comment (valfirst) [reply vividus-bdd-engine/src/test/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessorTests.java:58]: any suggestions how to make it better?
- `encodeUriXyz`
- `encodeUriXXX`
- `encodeUri***`
- just message
...

Comment (ikalinin1) [reply vividus-bdd-engine/src/test/java/org/vividus/bdd/expression/EncodeUrlExpressionProcessorTests.java:58]: encodeUri<part> ?

**Pull request merged**